hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -233,7 +233,7 @@ void doRemoveAddress(Address deadAddress, boolean destroyConnection) {
             }
             Connection conn = node.connectionManager.getConnection(deadAddress);
             if (destroyConnection && conn != null) {
-                node.connectionManager.destroyConnection(conn);
+                conn.close(reason, null);
```

Comment:
Hmm, it looks like `TcpIpConnection::close()` is calling `TcpIpConnectionManager::destroyConnection(this);` internally. Nevertheless we may skip to call `closedCount.inc();` now, since the connection may be already closed then.

**Before**
`TcpIpConnectionManager::destroyConnection()` is called
* executes some logic
* checks `TcpIpConnection::isAlive()`
* if `true` calls `TcpIpConnection::close()` and `closedCount.inc()`
* `TcpIpConnection::close()` is called
  * executes some logic
  * calls `TcpIpConnectionManager::destroyConnection()`
  * executes some logic
  * checks `TcpIpConnection::isAlive()`
  * should be `false` by now

**After**
`TcpIpConnection::close()` is called
* executes some logic
* calls `TcpIpConnectionManager::destroyConnection()`
  * executes some logic
  * checks `TcpIpConnection::isAlive()`
  * should be `false` by now, `TcpIpConnection::close()` and `closedCount.inc()` are skipped

Did I get that right?